src/services/nnas/routes/account-settings.ts

```diff
@@ -0,0 +1,228 @@
+	const server: HydratedServerDocument | null = await getServerByClientID('3f3928cc6f780638d360f0485cef973f', config.server_environment);
+	const token = getValueFromHeaders(request.headers, 'x-nintendo-service-token');
+
+	if (!server || !token) {
+		response.sendStatus(504);
+		return;
+	}
+
+	const aesKey = server?.aes_key;
+	const decryptedToken = decryptToken(Buffer.from(token, 'base64'), aesKey);
+	const tokenContents: Token = unpackToken(decryptedToken);
+
+	try {
+		const pnid: HydratedPNIDDocument | null = await getPNIDByPID(tokenContents.pid);
+		const personBody: AccountSettings = request.body;
+
+		if (!pnid) {
+			response.status(404);
+			response.redirect('/v1/account-settings/ui/profile');
+			return;
+		}
+
+		const person = accountSettingsSchema.safeParse(personBody);
+
+		if (!person.success) {
+			response.status(404);
+			response.redirect('/v1/account-settings/ui/profile');
+			return;
+		}
+
+		const timezoneName = (person.data.tz_name && !!Object.keys(person.data.tz_name).length) ? person.data.tz_name : pnid.timezone.name;
+
+		const regionLanguages: RegionLanguages = timezones[pnid.country as keyof typeof timezones];
+		const regionTimezones: RegionTimezones = regionLanguages[pnid.language] ? regionLanguages[pnid.language] : Object.values(regionLanguages)[0];
+		const timezone: RegionTimezone | undefined = regionTimezones.find(tz => tz.area === timezoneName);
+		const country: Country | undefined = regionsList.find((region) => region.iso_code === pnid.country);
+		let notice = '';
+
+		if (!country) {
+			response.status(404);
+			response.redirect('/v1/account-settings/ui/profile');
+			return;
+		}
+
+		const regionObject: Region | undefined = country.regions.find((region) => region.id === person.data.region);
+		const region = regionObject ? regionObject.id : pnid.region;
+
+		if (!timezone) {
+			response.status(404);
+			response.redirect('/v1/account-settings/ui/profile');
+			return;
+		}
+
+		pnid.birthdate = person.data.birthdate;
+		pnid.gender = person.data.gender;
+		pnid.region = region;
+		pnid.country = person.data.country;
+		pnid.timezone.name = timezoneName;
+		pnid.timezone.offset = Number(timezone.utc_offset);
+		pnid.flags.marketing = person.data.marketing_flag;
+		pnid.flags.off_device = person.data.off_device_flag;
+
+		if (person.data.server_selection) {
+			const environment = person.data.server_selection;
+
+			if (environment === 'test' && pnid.access_level < 1) {
+				response.status(400);
+				notice = 'Do not have permission to enter this environment';
+				response.redirect(`/v1/account-settings/ui/profile?notice=${notice}`);
+				return;
+			}
+
+			if (environment === 'dev' && pnid.access_level < 3) {
+				response.status(400);
+				notice = 'Do not have permission to enter this environment';
+				response.redirect(`/v1/account-settings/ui/profile?notice=${notice}`);
+				return;
+			}
```

Comment:
These access level checks remind me of the ones from the admin panel, which caused issues. I know there's basically no chance of that now, but better safe than sorry